scylla/src/transport/session.rs

```diff
@@ -1766,7 +1766,19 @@ impl Session {
 
         let partition_key = calculate_partition_key(prepared, serialized_values)?;
 
-        Ok(Some(partitioner_name.hash(&partition_key)))
+        Ok(Some((partitioner_name.hash(&partition_key), partition_key)))
+    }
+
+    pub fn calculate_token(
+        &self,
+        prepared: &PreparedStatement,
+        serialized_values: &SerializedValues,
+    ) -> Result<Option<Token>, QueryError> {
+        match self.calculate_token_and_routing_key(prepared, serialized_values) {
+            Ok(Some((token, _))) => Ok(Some(token)),
+            Ok(None) => Ok(None),
+            Err(err) => Err(err),
+        }
```

Comment:
I'd rather keep `calculate_token` as it is - a user recently sent a PR to make it public, so I assume they would like to keep this interface.

Reply:
Let me rephrase: the interface I propose is as follows:
```
pub fn calculate_token(
    &self,
    prepared: &PreparedStatement,
    serialized_values: &SerializedValues,
) -> Result<Option<Token>, QueryError> {
```
(unchanged)

and:
```
fn calculate_routing_key(
    &self,
    prepared: &PreparedStatement,
    serialized_values: &SerializedValues,
) -> Result<Option<Token>, QueryError> {
```

with `calculate_token` implementation as follows:
```
pub fn calculate_token(
    &self,
    prepared: &PreparedStatement,
    serialized_values: &SerializedValues,
) -> Result<Option<Token>, QueryError> {
    match self.calculate_routing_key(prepared, serialized_values) {
        Ok(Some(routing_key)) => {
             let partitioner_name = prepared.get_partitioner_name();
             Ok(Some(partitioner_name.hash(&routing_key)))
        }
        otherwise => otherwise,
    }
    ```

Reply:
Ok, makes sense.